SonarQube.TeamBuild.PreProcessor/PreprocessorObjectFactory.cs

```diff
@@ -10,15 +10,23 @@
 
 namespace SonarQube.TeamBuild.PreProcessor
 {
-    internal class PreprocessorObjectFactory : IPreprocessorObjectFactory
+    /// <summary>
+    /// Default implementation of the object factory interface that returns the
+    /// product implementations of the required classes
+    /// </summary>
+    /// <remarks>
+    /// Note: the factory is stateful and expects objects to be requested in the 
+    /// order they are used
+    /// </remarks>
```

Comment:
Changed from being a singleton because of the need to hold per-instance state. I've also added tests for this class.